scripts/stage-docs.sh

```diff
@@ -0,0 +1,162 @@
+REPOSITORY="origin"
+BRANCH_FROM="main"
+DRY_RUN="false"
+ORIGINAL_GH_DEFAULT_REPO=""
+while getopts ":v:u:b:r:dh" opt; do
+  case $opt in
+    v) RELEASE_VERSION="${OPTARG}"
+    ;;
+    u) WEBSITE_REPO_URL="${OPTARG}"
+    ;;
+    b) BRANCH_FROM="${OPTARG}"
+    ;;
+    r) REPOSITORY="${OPTARG}"
+    ;;
+    d) DRY_RUN="true"
+    ;;
+    h)
+      1>&2 cat << EOF
+usage: $0 -v version -u url [-b branch] [-r repository] [-d] [-h]
+ -v version number e.g. 0.3.0
+ -u url of the website repository e.g. git@github.com:kroxylicious/kroxylicious.github.io.git
+ -b branch to release from (defaults to 'main')
+ -r the remote name of the kroxylicious.github.io repository (defaults to 'origin')
+ -d dry-run mode
+ -h this help message
+EOF
+      exit 1
+    ;;
+    \?) echo "Invalid option -$opt ${OPTARG}" >&2
+    exit 1
+    ;;
+  esac
+
+done
+
+if [[ -z ${RELEASE_VERSION} ]]; then
+  echo "No version specified, aborting"
+  exit 1
+fi
+
+if [[ -z ${WEBSITE_REPO_URL} ]]; then
+  echo "No website repository URL specified, aborting"
+  exit 1
+fi
+
+BLUE='\033[0;34m'
+NC='\033[0m' # No Color
+
+ORIGINAL_WORKING_BRANCH=$(git branch --show-current)
+
+cleanup() {
+    if [[ -n ${ORIGINAL_WEBSITE_WORKING_BRANCH} ]]; then
+        git checkout "${ORIGINAL_WEBSITE_WORKING_BRANCH}" || true
+    fi
+
+    if [[ ${ORIGINAL_GH_DEFAULT_REPO} ]]; then
+      gh repo set-default "${ORIGINAL_GH_DEFAULT_REPO}"
+    fi
+
+    # Note that git branch -D echos the sha of the deleted branch to
+    # stdout.  This is great for debugging the release process as it
+    # lets the developer restore to the state of the tree.
+    if [[ "${DRY_RUN:-false}" == true && ${RELEASE_DOCS_BRANCH} ]]; then
+        git branch -D "${RELEASE_DOCS_BRANCH}" || true
+    fi
+
+    cd ../kroxylicious/
+
+    if [[ -n ${ORIGINAL_WORKING_BRANCH} ]]; then
+        git checkout "${ORIGINAL_WORKING_BRANCH}" || true
+    fi
+}
+
+trap cleanup EXIT
+
+git stash --all
+
+RELEASE_DATE=$(date -u '+%Y-%m-%d')
+RELEASE_TAG="v${RELEASE_VERSION}"
+RELEASE_DOCS_BRANCH="prepare-${RELEASE_TAG}-release-docs-${RELEASE_DATE}"
+
+# Use a `/.` at the end of the source path to avoid the source path being appended to the destination path if the `.../_files/` folder already exists
+KROXYLICIOUS_DOCS_LOCATION="../kroxylicious/docs/."
+WEBSITE_DOCS_LOCATION="./docs/${RELEASE_TAG}"
+
+if [[ "${DRY_RUN:-false}" == true ]]; then
+    #Disable the shell check as the colour codes only work with interpolation.
+    # shellcheck disable=SC2059
+    printf "${BLUE}Dry-run mode:${NC} no remote tags or PRs will be created, artefacts will be deployed to: ${DEPLOY_DRY_RUN_DIR}\n"
+    GIT_DRYRUN="--dry-run"
+fi
+
+echo "Checking out tags/${RELEASE_TAG} in  in $(git remote get-url "${REPOSITORY}")"
+git checkout "tags/${RELEASE_TAG}"
+
+# Move up a directory so we don't end up with website files in the main repository
+cd ../
+echo "In '$(pwd)', cloning website repository at ${WEBSITE_REPO_URL}"
+git clone "${WEBSITE_REPO_URL}"
```

Comment:
Would it be better to clone the WEBSITE_REPO_URL in a temporary location?

Reply:
I wasn't sure if we have access to `/tmp` on the GitHub-hosted runner image we use (I seem to recall that being a problem for _something_, but I'm starting to think it probably wasn't anything to do with GitHub...), so I elected to use a directory I knew we should have access to.
I've had a look around on the internet and I can't see why we wouldn't be able to use `/tmp` (we'd just have to keep track of the full paths for everything, rather than using relative paths), so I can change this.

Reply:
I've changed this now